### PR TITLE
Generate a WordPress template

### LINKED ISSUE #5: Investigate previews
Writers need to preview blog posts before publishing them. Given that we are using WordPress as a headless CMS, we cannot use the default preview feature by default (not without some changes).

Some ideas:

- Have the WP theme consume the same styles from the blog and provide a themed header and footer so it’s clear what things will look like.
- Build a dynamic preview app that can build a preview using the same styles and templates that 11ty would use to build the blog.
- Look at plugins to help with this. For example the headless-cms plugin seems to provide a way to expose a header and footer via the API.

### PR DESCRIPTION
**Pull request by willdurand**
fixes #5

### CODE REVIEW

File: eleventy.config.js [hunk 1]
```diff
@@ -47,14 +50,11 @@ module.exports = function (eleventyConfig) {
...
 
-  eleventyConfig.addFilter(
-    'postsWithCategory',
-    (allPosts, category) => {
-      return allPosts.filter((item) => {
-        return item.categories.includes(category.id);
-      });
-    }
-  );
+  eleventyConfig.addFilter('postsWithCategory', (allPosts, category) => {
```
Comment (willdurand) [inline eleventy.config.js:53]: (Prettier did that)

File: src/wp-content/blogposts.njk [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        </div>
+      </div>
+      <aside class="Sidebar">
+        <dl class="Meta">
+          <dt class="published">Updated</dt>
+          <dd class="published"><?= the_time() ?></dd>
+          <dt>Author</dt>
+          <dd>
+            <ul class="Authorlist">
+              <li class="Author"><?= the_author() ?></li>
```
Comment (willdurand) [inline src/wp-content/blogposts.njk:25]: Not sure why but the author name seems to be displayed twice in the preview.

**Comment by willdurand** [comment]
@muffinresearch when you have a minute or two, take a look at our WP instance. It is a barebone theme but it is quite similar to the generated site we have and the added code isn't super complex. We might be able to create more partials/includes in order to no repeat some parts of the layout, although it might be tricky given that we use filters for dates, etc.

File: src/wp-content/index.njk [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+{% block content %}
+  <ul class="archive">
+    <?php while (have_posts()) : the_post(); ?>
+      <li>
+        <p class="">
+          <time class="date" datetime=""></time>
+        </p>
+        <h2 class="post-title">
+          <a class="" href="" class="postlist-link">
```
Comment (willdurand) [inline src/wp-content/index.njk:16]: TODO: add the URL to the blog post

File: src/assets/css/header.scss [hunk 4]
```diff
@@ -182,7 +182,7 @@
   $largeHeight: 64px;
   $largeWidth: $largeHeight * $aspectRatio;
 
-  background-image: url('/assets/img/logo.svg');
```
Comment (muffinresearch) [inline src/assets/css/header.scss:185]: As a heads-up not starting a path with '/assets/' will break the hashed filename rewriting in the build script unless it's updated to handle relative paths.

File: src/wp-content/blogposts.njk (hunk 2 above)
Comment (muffinresearch) [reply src/wp-content/blogposts.njk:25]: I looked at the user, in-case it was related to not having a first and last name but that made no difference.

Comment (muffinresearch) [reply src/wp-content/blogposts.njk:25]: Looks like you might want https://developer.wordpress.org/reference/functions/get_the_author/ instead. `the_author()` echos and returns the author by default.

Comment (willdurand) [reply src/wp-content/blogposts.njk:25]: ah, indeed. I didn't pay attention to that, I inferred the function's name based on the others.

Comment (willdurand) [reply src/wp-content/blogposts.njk]: fixed

File: src/wp-content/index.njk (hunk 3 above)
Comment (willdurand) [reply src/wp-content/index.njk]: fixed

File: src/assets/css/header.scss (hunk 4 above)
Comment (willdurand) [reply src/assets/css/header.scss:185]: I suppose this happens in the `asset-pipeline` script that we don't yet have in this repo?

**Review by muffinresearch — APPROVED** [review]
r+

**Pull request merged**